### PR TITLE
Update annotate-snippets-rs to 0.8.0

### PR DESCRIPTION
**Pull request by zzau13**
#59346
I made major changes to this library. In the previous version we worked with owned while in the current one with borrowed.

I have adapted it without changing the behavior.
I have modified the coverage since the previous one did not return correctly the index of the character in the line.

### CODE REVIEW

File: src/librustc_errors/emitter.rs [hunk 1]
```diff
@@ -1379,7 +1379,6 @@ impl EmitterWriter {
                 for line in &annotated_file.lines {
                     for ann in &line.annotations {
                         span_left_margin = min(span_left_margin, ann.start_col);
-                        span_left_margin = min(span_left_margin, ann.end_col);
```
Comment (estebank) [inline src/librustc_errors/emitter.rs:1382]: What do these two changes do?

**Review by estebank — COMMENTED** [review]
Took a cursory look and it LGTM, but would like to dig deeper just to check. I also noticed a small change that I'm unsure about.

File: src/librustc_errors/emitter.rs (hunk 1 above)
Comment (botika) [reply src/librustc_errors/emitter.rs:1382]: Sorry, I was fine. I got confused start and end and didn't see the column. Now I turn it back.

**Comment by Dylan-DPC** [comment]
merging this

@bors r=estebank

**Comment by RalfJung** [comment]
Yielding to rollup containing this
@bors retry

**Pull request merged**